.github/workflows/pr-realm-js.yml

```diff
@@ -189,68 +214,172 @@ jobs:
           echo 'UseMultiToolTask=true' >> $GITHUB_ENV
           echo 'VCPKG_KEEP_ENV_VARS=CCACHE_COMPILER;CCACHE_STATSLOG;CCACHE_LOGFILE;CCACHE_COMPILERTYPE;UseMultiToolTask' >> $GITHUB_ENV
 
-      # build the c++ library for standard targets
-      - name: Build node
-        if: ${{ matrix.variant.os != 'ios' && matrix.variant.os != 'android' }}
-        run: npm run build:node:prebuild:${{matrix.variant.arch}} --workspace realm
-
-      # build the c++ library for IOS
-      # the Info.plist needs to be regenerated with all libraries in place
-      - name: Build iOS
-        if: ${{ matrix.variant.os == 'ios' }}
-        run: |
-          npm run build:ios --workspace realm
-          rm -vf ${{ matrix.variant.artifact-path }}/Info.plist
+      - name: Build native module
+        run: npm run prebuild-node --workspace realm
         env:
-          PLATFORMS: ${{ matrix.variant.arch }}
-
-      # build the c++ library for Android
-      - name: Build Android
-        if: ${{ matrix.variant.os == 'android' }}
-        run: npm run build:android --workspace realm -- --arch=${{matrix.variant.arch}}
+          PREBUILD_ARCH: ${{ matrix.arch }}
 
-      # Due to a limitation in upload-artifact a redundant file is needed to force
-      # preserving paths (https://github.com/actions/upload-artifact/issues/174)
       - name: Upload prebuild artifact
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: realm-js-prebuilds
+          name: node-${{ matrix.os }}-${{ matrix.arch }}-prebuild
+          # Including README.md to pin paths to the root of the repository
           path: |
             README.md
-            ${{ matrix.variant.artifact-path }}
+            packages/realm/prebuilds/realm-*-napi-*.tar.gz
 
-  ios-xcframework:
-    name: Generate Info.plist with all frameworks in place
-    needs: [build]
-    if: ${{ success() || failure() }}
-    runs-on: macos-latest
+  prebuild-apple-archives:
+    name: Prebuild ${{ matrix.platform }} (Apple React Native)
+    runs-on: macos-latest-large
```

Comment:
Do we need a large runner?

Reply:
I'm experimenting with the `large` ... it's currently `xlarge` https://github.com/realm/realm-js/blob/b121047ac34e632bac452d1627ab9b8734929018/.github/workflows/pr-realm-js.yml#L77-L79

We might be able to get smaller, but I'd like to experiment with that separately from this PR 👍

Reply:
Sounds good to experiment later